charts/redpanda/templates/statefulset.yaml

```diff
@@ -253,18 +303,14 @@ spec:
             - --memory={{ template "redpanda-memory" . }}M
             - --reserve-memory={{ template "redpanda-reserve-memory" . }}
             - --default-log-level={{ .Values.logging.logLevel }}
-            - --advertise-rpc-addr={{ $advertiseAddress }}:{{ .Values.listeners.rpc.port }}
-            - --advertise-pandaproxy-addr=internal://{{ $advertiseAddress }}:{{ .Values.listeners.http.port }},
-{{- range $name, $listener := .Values.listeners.http.external -}}
-                {{ $name }}://{{ $advertiseAddress }}:{{ $listener.nodePort }},
-{{- end }}
+            - --advertise-rpc-addr={{ $internalAdvertiseAddress }}:{{ .Values.listeners.rpc.port }}
           ports:
 {{- range $name, $listener := .Values.listeners }}
-            - name: {{ lower $name }}
+            - name: {{ lower $name | trunc 6 }}-internal
               containerPort: {{ $listener.port }}
   {{- range $externalName, $external := $listener.external }}
     {{- if $external.port }}
-            - name: {{ lower $name | trunc 6 }}-{{ lower $externalName | trunc 8}}
+            - name: {{ lower $name | trunc 4 }}-ext-{{ lower $externalName | trunc 6 }}
```

Comment:
Isn't internal already named "internal"? I'd prefer we not truncate these any more than we already have. If the default names are not descriptive enough (and "default" may be a bit too generic), let's change those.

Reply:
I didn't want to truncate here, but it is required that port names are 15 or less characters. The deployment will fail otherwise. So I truncated the name in the best way I could figure out (to keep defails from each part of the name) to fit within the 15 character restriction. At one point I found the kubernetes docs for this, but just now I couldn't find the link again. But incrementing either the 4 or 6 (or just adding in a single character anywhere) will show you the error.

Reply:
Right, but aren't we truncating only because we're changing the names and adding extra characters?

Reply:
Ahh, you are more asking why I have made this modification in this PR rather than asking why `trunc` is needed.

The change is not strictly needed, I just realized while developing this PR that we did not explicitly name the internal port with anything that mentioned `internal` (and same for external ports).

I'll remove this change to keep the PR more focused.